crates/cargo-subcommand-metadata/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+#[cfg(feature = "parse")]
+pub mod parse;
+
+/// Cargo's name for the purpose of ELF notes.
+///
+/// The `name` field of an ELF note is designated to hold the entry's "owner" or
+/// "originator". No formal mechanism exists for avoiding name conflicts. By
+/// convention, vendors use their own name such as "XYZ Computer Company".
+pub const ELF_NOTE_NAME: &str = "rust-lang/cargo";
+
+/// Values used by Cargo as the `type` of its ELF notes.
+///
+/// Each originator controls its own note types. Multiple interpretations of a
+/// single type value can exist. A program must recognize both the `name` and
+/// the `type` to understand a descriptor.
+#[repr(i32)]
+#[non_exhaustive]
+pub enum ElfNoteType {
+    //              DESCRIP
+    Description = 0xDE5C819,
+}
+
+/// Embed a description into a compiled Cargo subcommand, to be shown by `cargo
+/// --list`.
+///
+/// The following restrictions apply to a subcommand description:
+///
+/// - String length can be at most 280 bytes in UTF-8, although much shorter is
+///   better.
+/// - Must not contain the characters `\n`, `\r`, or `\x1B` (ESC).
+///
+/// Please consider running `cargo --list` and following the style of the
+/// existing descriptions of the built-in Cargo subcommands.
+///
+/// # Example
+///
+/// ```
+/// // subcommand's main.rs
+///
+/// cargo_subcommand_metadata::description! {
+///     "Draw a spiffy visualization of things"
+/// }
+///
+/// fn main() {
+///     /* … */
+/// }
+/// ```
+#[macro_export]
+macro_rules! description {
+    ($description:expr) => {
```

Comment:
Note to self: maybe `cargo_subcommand_metadata::description!();` with no string argument could embed `env!(CARGO_PKG_DESCRIPTION)` by default.